pymeasure/adapters/adapter.py

```diff
@@ -34,44 +37,112 @@ class Adapter:
     This class should only be inherited from.
 
     :param preprocess_reply: optional callable used to preprocess strings
-        received from the instrument. The callable returns the processed string.
+        received from the instrument. The callable returns the processed
+        string. This argument is deprecated.
```

Comment:
For all deprecations, we should make sure to include the word deprecated/deprecation somewhere relevant (this makes it easier to search for those in the codebase).
We should also add an issue to _remove_ the deprecated features so that we don't forget when it's time (e.g. before v1.0).

Reply:
I'd put the deprecation front-and-center here, so it's harder to miss: 
`:param preprocess_reply: (deprecated) Optional callable...`

Reply:
Even better -- there is a `deprecated` Sphinx directive: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated

Reply:
The directive does not work nicely on parameters, but for methods it works great. Thanks.

Reply:
Does [this ](https://stackoverflow.com/a/70593876/599884)work for you, or is there some drawback I don't see without actually creating an example?

Reply:
It works.